File: ctrlm/app_siret/sge.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jul 13 01:51:12 2020

@author: C21373
"""
from webcontroller.ie import IExplore
import models
import sys

URL_SGE_ROOT = "https://sge.enedis.fr"
URL_PDL = "/sgePortail/appmanager/portail/bureau?_nfpb=true&_windowLabel=portlet_gestion_pdl_2&portlet_gestion_pdl_2_actionOverride=%2Fportlets%2FgestionPdl%2FconsulterPdm&portlet_gestion_pdl_2idPdm=[PDL]&_pageLabel=sge_page_accueil"


def readPdl(rae):
    ie= IExplore("Système de Gestion des Echanges")
    if ie == None:
        print(" Echèc  control internet explorer")
    ie.load(URL_SGE_ROOT + URL_PDL.replace('[PDL]',"{:>014}".format(rae)))

    try:
        tpl_rae = ((ie.getdocument().all("prmId").value), 
                    (ie.getdocument().all("voie").value),
                    (ie.getdocument().all("codePostal").value),
                    (ie.getdocument().all("commune").value),
                    (ie.getdocument().all("codeINSEE").value)
                )                   
    except:
        if (ie.getwebbrowser().locationUrl == URL_SGE_ROOT + URL_PDL.replace('[PDL]',rae)):
            tpl_rae =(rae, "pdl inexistant", "", "", "", )
        else:
            tpl_rae = ()
            print("erreur sur le pdl: " , rae )
    
    return tpl_rae

todolist = models.readMinus("liste", "pdl", 'rae,siren')
print(len(todolist))


for tpl_row in todolist:
    tpl_rae = readPdl(tpl_row[0])
    tpl_raesiren = tpl_rae + (tpl_row[1],)
    
    header = ("rae","voie","code_postal","commune","code_insee","siren" )
    if len(tpl_raesiren)  == len(header):
       try:
           models.write("pdl", header, tpl_raesiren)
       except:
           print("Unexpected error:", sys.exc_info())        
               
    
    print(tpl_raesiren)

File: templates/scripts/sge.js
const TITLE_SGE = "Système de Gestion des Echanges";
const URL_SGE = "https://sge.enedis.fr/sgePortail/appmanager/portail/bureau?" + 
                "_nfpb=true&_windowLabel=portlet_gestion_pdl_2&portlet_gestion_pdl_2_actionOverride=%2Fportlets%2FgestionPdl%2FconsulterPdm&portlet_gestion_pdl_2idPdm=[PDL]&_pageLabel=sge_page_accueil"; 

function FichePrm(noRAE){
    this.noRAE = noRAE;
    this.domFicheSGE;
    this.addressSGE;
    this.isDisplay = function (){
        return  document.getElementById( this.noRAE)
    }
    this.readAddressPrm = function(){
            this.addressSGE = {
                                pdl_id:this.domFicheSGE.all("prmId").value,
                                pdl_voie: this.domFicheSGE.all("voie").value, 
                                pdl_cp: this.domFicheSGE.all("codePostal").value, 
                                pdl_commune: this.domFicheSGE.all("commune").value
                                };
            return  this.addressSGE;      
    try {
    
        } 
        catch (error) {
          console.error("la page sge n'est pas correct");
          return  false;
        }
    }

    this.getUrlFicheSGE = function(){
        return URL_SGE.replace("[PDL]", this.noRAE)
    }

    this.loadFicheSGE = function (){
        if (this.addressSGE) return true;

            const that = this;		
            callAjax(
                "POST", 
                this.getUrlFicheSGE(),
                function(o_xhr){
                    if (o_xhr.response.title === TITLE_SGE) {
                        that.domFicheSGE = o_xhr.response;
                        that.readAddressPrm();
                        that.DisplayPDL();
                    
                    }  

                    else
                    {
                       console.log("Error connexion: la page chargée n'est celle qui est attendue " + o_xhr.response.title) 
                        //setTimeout(10000, callSGE(this.noRAE))
                    }
                }, 
                function(o_xhr){
                    console.log("Error connexion: pas de reponse") ;
                }
            );				
    }

    this.DisplayPDL = function(){
        try{
            const o_htmlRowElement = document.getElementById(  this.addressSGE.pdl_id); 
            HTMLTableRowElement.display(o_htmlRowElement,this.addressSGE)
                                   /* [
                                        ["pdl_voie", this.addressSGE.pdl_voie], 
                                        ["pdl_cp", this.addressSGE.pdl_cp],
                                        ["pdl_commune", this.addressSGE.pdl_commune]
                                    ]); */
        } catch(error){
            console.log("error : pas d'info d'adresse disponible");
        }
    }

}

/*
function StringRepeat(str, nombre){
    const arr = new Array();
    for(let i=0; i<nombre; i++){
        arr.push(str);
    } 
    return  arr.join("");
}

}*/